src/util/NamedColor.h

```diff
@@ -0,0 +1,136 @@
+/**
+ * @file NamedColor.h
+ * @author idotobi
+ * @brief Struct for NamedColor
+ *
+ */
+
+#pragma once
+#include <string>
+
+#include <util/Color.h>
+
+struct Palette;
+/**
+ * @brief NamedColor is an object for a color with name and an optional index which refers to the index of the color
+ * inside a given palette.
+ * If the NamedColor does not come from a Color palette isColorPalette is set to false and the palettIndex is 0.
+ *
+ */
+struct NamedColor {
+    /**
+     * @brief Default Construct a new Named Color instance
+     * This is only needed for the input operator implementation
+     *
+     */
+    NamedColor();
+
+    /**
+     * @brief Copy Construct a new NamedColor instance
+     *
+     * @param other NamedColor instance to copy from
+     */
+    NamedColor(const NamedColor& other);
+
+    /**
+     * @brief Copy assignment operator
+     *
+     * @param other NamedColor instance to copy from
+     * @return NamedColor& NamedColor instance to copy to
+     */
+    NamedColor& operator=(const NamedColor& other);
+
+    /**
+     * @brief Move Construct a new NamedColor instance
+     *
+     * @param other NamedColor instance to move from
+     */
+    NamedColor(NamedColor&& other) noexcept;
+
+    /**
+     * @brief Move assignment  opearator
+     *
+     * @param other NamedColor instance to move from
+     * @return NamedColor& NamedColor instance to move to
+     */
+    NamedColor& operator=(NamedColor&& other) noexcept;
+
+    /**
+     * @brief Construct a new NamedColor instance
+     * This automatically sets the isPaletteColor to negative as NamedColors from a color palette can only be
+     * constructed using the input operator.
+     * As it is no paletteColor the Name of the Color will be set to "Custom Color" and the paletteIndex to 0.
+     *
+     * @param color color to set for the custom NamedColor instance
+     */
+    NamedColor(const Color& color);
+
+    ~NamedColor();
```

Comment:
```suggestion
```

Reply:
Alternatively mark all as defaulted.
In my opinion, it is not required to write documentation for trivial constructors / destructors.

Reply:
I'll remove them (see above).